### ISSUE TITLE
[Grid - Selection] Missing selected state for single select

### ISSUE THREAD

**Issue by Magsu** [sl-design-system/components#2896]
### Provide a general summary of the issue here

Currently the single select variant is missing a selected / not selected state for the rows. The multiselect variant uses a checkbox, but the single select uses a button. Users should be aware that they can only select a single row, and assistive technology users should know when an item is selected or not. 

<img width="1876" height="310" alt="Single select grid" src="https://github.com/user-attachments/assets/a1c3f32a-5e7b-4f19-a918-3b4c38e7abd1" />

### 🤔 Expected Behavior?

When user enters the table, they are aware that they can only select one at a time. Assistive technology users are aware when an item is selected and when not. 

### 😯 Current Behavior

Assistive technology users are aware, that the items are buttons, but when activating the button, nothing indicates what the action was, if anything even happened. 

### 💁 Possible Solution

As users can select and deselect the items, these are closes to either an exclusive checkbox or switch/toggle button group.  Anyways, they should communicate whether they are selected or not. 

### 🖥️ Steps to Reproduce

1. Navigate to a row using screen reader
2. Activate the row by using space/enter -> Screen reader stays silent (or repeats the button name & role depending on sr)

### What browsers are you seeing the problem on? (only for bugs in code)

Chrome

### What operating system are you using?

Windows 11

### 👤 Your name

Valtteri Järvinen

### 🧢 Your product/team

External

### 🎨 Your Theme(s)

Sanoma Learning

**Comment by Diaan** [comment]
use aria-selected
